crates/validation/src/order/mod.rs

```diff
@@ -259,4 +266,16 @@ impl OrderValidatorHandle for ValidationClient {
             rx.await.unwrap()
         })
     }
+
+    fn estimate_gas(&self, order: AllOrders) -> GasEstimationFuture {
+        Box::pin(async move {
+            match self.validate_order(OrderOrigin::External, order).await {
+                OrderValidationResults::Valid(o) => Ok(o.priority_data.gas_units),
+                OrderValidationResults::Invalid(e) => Err(format!("Invalid order: {}", e)),
+                OrderValidationResults::TransitionedToBlock => {
+                    Err("Order transitioned to block".to_string())
+                }
+            }
+        })
+    }
```

Comment:
We want the gas, converted to erc-20 tokens aswell. This way we can also derive the rate of erc-20 token per gas. This is because we charge gas on chain in erc-20 based on this conversion rate thats calculated